projects/core/src/routing/event/routing.events.ts

```diff
@@ -0,0 +1,70 @@
+import { Images } from '../../model';
+import {
+  BaseOption,
+  Category,
+  Classification,
+  FutureStock,
+  Price,
+  PriceRange,
+  Product,
+  ProductReferences,
+  Promotion,
+  Review,
+  Stock,
+  VariantMatrixElement,
+  VariantOption,
+  VariantType,
+} from '../../model/product.model';
+
+export class PageVisited {}
+export class HomePageVisited extends PageVisited {}
+export class CartPageVisited extends PageVisited {}
+export class OrderConfirmationPageVisited extends PageVisited {}
+
+/**
+ * Indicates that a user visited a PDP page.
+ * A visited product code value is emited whenever the PDP page is visited.
+ */
+export class ProductDetailsPageVisited extends PageVisited implements Product {
```

Comment:
This seems a bit too much to be considered as a generic PDP visit event.
Let's try to figure out some properties that would be useful to everyone:
- `code`
- `name`
- `price`
- `purchasable` - what's this? :)
- `categories`

Do you think you need something else?

Reply:
These are all we need, well minus purchasable... not sure what that is